### PR TITLE
[search] HotelsClassifier can be used on the UI side now.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mpimenov @alexzatsepin @igrechuhin PTAL

### CODE REVIEW

File: search/result.hpp [hunk 1]
```diff
@@ -133,67 +133,63 @@ class Result
...
-    res.SetPositionInResults(static_cast<int32_t>(m_vec.size()));
-    m_vec.push_back(move(res));
+    m_status = cancelled ? STATUS_ENDED_CANCELLED : STATUS_ENDED_NORMAL;
   }
 
-  inline void Clear() { m_vec.clear(); }
+  bool AddResult(Result && result);
 
-  typedef vector<Result>::const_iterator IterT;
+  // Fast version of AddResult() that don't do any duplicates checks.
```
Comment (mpimenov) [inline search/result.hpp]: s/don't/doesn't/

Comment (ygorshenin) [inline search/result.hpp]: Done

File: search/hotels_classifier.hpp [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+#include "search/result.hpp"
 
 namespace search
 {
-class Results;
-
-// A binary classifier that can be used in conjunction with search
-// engine to decide whether the majority of results are hotels or not.
-//
+// A binary classifier that can be used in conjunction with search
```
Comment (ygorshenin) [inline search/hotels_classifier.hpp]: Done

**Comment by ygorshenin** [comment]
PTAL

**Comment by mpimenov** [comment]
LGTM

**Comment by igrechuhin** [comment]
LGTM

**Comment by alexzatsepin** [comment]
LGTM

**Pull request merged**